src/lib/jellyfin-apiclient/connectionManager.js

```diff
@@ -0,0 +1,760 @@
+                throw new Error(`Server not found: ${serverId}`);
+            }
+
+            const server = servers[0];
+
+            return self._getOrAddApiClient(server, getServerAddress(server, server.LastConnectionMode));
+        };
+
+        function onAuthenticated(apiClient, result, options, saveCredentials) {
+            const credentials = credentialProvider.credentials();
+            const servers = credentials.Servers.filter((s) => s.Id === result.ServerId);
+
+            const server = servers.length ? servers[0] : apiClient.serverInfo();
+
+            if (options.updateDateLastAccessed !== false) {
+                server.DateLastAccessed = new Date().getTime();
+            }
+            server.Id = result.ServerId;
+
+            if (saveCredentials) {
+                server.UserId = result.User.Id;
+                server.AccessToken = result.AccessToken;
+            } else {
+                server.UserId = null;
+                server.AccessToken = null;
+            }
+
+            credentialProvider.addOrUpdateServer(credentials.Servers, server);
+            credentialProvider.credentials(credentials);
+
+            // set this now before updating server info, otherwise it won't be set in time
+            apiClient.enableAutomaticBitrateDetection = options.enableAutomaticBitrateDetection;
+
+            apiClient.serverInfo(server);
+            apiClient.setAuthenticationInfo(result.AccessToken, result.User.Id);
+            afterConnected(apiClient, options);
+
+            return onLocalUserSignIn(server, apiClient.serverAddress(), result.User);
+        }
+
+        function afterConnected(apiClient, options = {}) {
+            if (options.reportCapabilities !== false) {
+                apiClient.reportCapabilities(capabilities);
+            }
+            apiClient.enableAutomaticBitrateDetection = options.enableAutomaticBitrateDetection;
+
+            if (options.enableWebSocket !== false) {
+                console.log('calling apiClient.ensureWebSocket');
+
+                apiClient.ensureWebSocket();
+            }
+        }
+
+        function onLocalUserSignIn(server, serverUrl, user) {
+            // Ensure this is created so that listeners of the event can get the apiClient instance
+            self._getOrAddApiClient(server, serverUrl);
+
+            // This allows the app to have a single hook that fires before any other
+            const promise = self.onLocalUserSignedIn ? self.onLocalUserSignedIn.call(self, user) : Promise.resolve();
+
+            return promise.then(() => {
+                events.trigger(self, 'localusersignedin', [user]);
+            });
+        }
+
+        function validateAuthentication(server, serverUrl) {
+            return ajax({
+                type: 'GET',
+                url: `${serverUrl}/System/Info`,
+                dataType: 'json',
+                headers: {
+                    'X-MediaBrowser-Token': server.AccessToken
```

Comment:
https://gist.github.com/nielsvanvelzen/ea047d9028f676185832e51ffaf12a6f :-)

Last time I checked this is the only part in jellyfin-web using a deprecated header so worth fixing now.